raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1635,24 +1693,29 @@ private Optional<Boolean> maybeHandleCommonResponse(
     }
 
     private void maybeTransition(
-        OptionalInt leaderId,
+        Optional<Node> leader,
         int epoch,
         long currentTimeMs
     ) {
+        OptionalInt leaderId = OptionalInt.empty();
```

Comment:
Again would be more clearly expressed with `?:` rather than mutating the OptionalInt reference

Reply:
Fixed.